src/coreclr/inc/corinfo.h

```diff
@@ -344,16 +344,93 @@ enum StructFloatFieldInfoFlags
     STRUCT_FLOAT_FIELD_SECOND     = 0x4,
     STRUCT_FIRST_FIELD_SIZE_IS8   = 0x10,
     STRUCT_SECOND_FIELD_SIZE_IS8  = 0x20,
+};
+
+// Bitfields for FpStructInRegistersInfo::flags
+namespace FpStruct
+{
+    enum class IntKind
```

Comment:
Why does the JIT need to know this detail? I assume that the JIT has map back the offset to the field to figure out the actual integer type (int8/int16/int32, etc.). Is that right?

Reply:
I put it in because recording it was cheap and it avoided calling getClassGClayout in morph.cpp on potentionally unbounded structures (in case of RISC-V). I see now JIT has a wrapper for it (ClassLayout) so it's less of a concern.

Note that `CorInfoType` of `loweredElements` will contain the same (even more) detail.

Reply:
Removed, there was one more place where I used it in reconstructing the returned struct after CallDescrWorker, but I think we can also manage without it.